src/lib.rs

```diff
@@ -598,6 +611,7 @@ impl Rng {
 }
 
 /// Run an operation with the current thread-local generator.
+#[cfg(feature = "std")]
 #[inline]
 fn with_rng<R>(f: impl FnOnce(&mut Rng) -> R) -> R {
```

Comment:
I'm hesitant to expose `(try_)with_rng`, since I feel like the API is much more idiomatic without giving direct access to the thread-local RNG. Anyways, I moved the thread-local things into a `global_rng` module.